src/db/collections/outputs/mod.rs

```diff
@@ -320,8 +320,8 @@ impl MongoDb {
                 .unwrap_or_default();
 
             Ok(Some(BalancesResult {
-                total_balance: balances.total_balance.amount as u64,
-                sig_locked_balance: balances.sig_locked_balance.amount as u64,
+                total_balance: balances.total_balance[0].amount as u64,
```

Comment:
You should be able to remove the `Balances` struct, and instead directly deserialize a `BalancesResult`, by using `$first` in a `$project`.

Reply:
Thanks, Alex. I bet you're right on this. Will try to do it a bit later.